In [1]:
%reload_ext autoreload
%autoreload 2

##### Define ChEMBL and Papyrus versions

In [2]:
chembl_version = '31'
papyrus_version = '05.5'
papyrus_flavor = 'nostereo'

##### Define annotation round

In [3]:
annotation_round = 2 # Use only for semi-manual annotation rounds (e.g. round 2)
previous_annotation_round = annotation_round - 1

##### Define data directory

In [4]:
from mutants_in_pcm import data_path
data_path.data_dir = '../data'

##### Import analysis functions

In [18]:
import numpy as np
import pandas as pd
import ast

from mutants_in_pcm.preprocessing import obtain_chembl_data,combine_chembl_papyrus_mutants,merge_chembl_papyrus_mutants
from mutants_in_pcm.annotation import *
from mutants_in_pcm.annotation_check import (read_assay_annotations,read_manual_positive_annotations,
                                             classify_negative_annotations,check_manual_positive_annotations,filter_positive_annotations,filter_negative_annotations)

#### Explore and check automatic annotation from the previous round

In [19]:
# Read assays with previous round automatic annotations
assays_previous_round = read_assay_annotations(chembl_version, previous_annotation_round)
assays_previous_round

assay_id                                        description accession  \
0          83907  In vivo inhibitory activity against human Hepa...    Q9Y251   
1         154606  Inhibitory activity against Palmitoyl-CoA oxid...    P07872   
2          51352  Inhibition of cytochrome P450 1A2 of isolated ...    P05177   
3          51895  Inhibition of cytochrome P450 3A4 of isolated ...    P08684   
4          51528  Inhibition of cytochrome P450 2C9 of isolated ...    P11712   
...          ...                                                ...       ...   
376228   2136225              KinomeScan assay: inhibition of LATS2    Q9NRM7   
376229   2136385              KinomeScan assay: inhibition of ROCK1    Q13464   
376230   2136386              KinomeScan assay: inhibition of ROCK2    O75116   
376231   2136430             KinomeScan assay: inhibition of TGFBR2    P37173   
376232   2136450               KinomeScan assay: inhibition of ULK1    O75385   

                                                 sequence mutation aa_change  \
0       MLLRSKPALPPPLMLLLLGPLGPLSPGALPRPAQAQDVVDLDFFTQ...      NaN        []   
1       MNPDLRKERASATFNPELITHILDGSPENTRRRREIENLILNDPDF...      NaN        []   
2       MALSQSVPFSATELLLASAIFCLVFWVLKGLRPRVPKGLKSPPEPW...      NaN        []   
3       MALIPDLAMETWLLLAVSLVLLYLYGTHSHGLFKKLGIPGPTPLPF...      NaN        []   
4       MDSLVVLVLCLSCLLLLSLWRQSSGRGKLPPGPTPLPVIGNILQIG...      NaN        []   
...                                                   ...      ...       ...   
376228  MRPKTFPATTYSGNSRQRLQEIREGLKQPSKSSVQGLPAGPNSDTS...      NaN        []   
376229  MSTGDSFETRFEKMDNLLRDPKSEVNSDCLLDGLDALVYDLDFPAL...      NaN        []   
376230  MSRPPPTGKMPGAPETAPGDGAGASRQRKLEALIRDPRSPINVESL...      NaN        []   
376231  MGRGLLRGLWPLHIVLWTRIASTIPPHVQKSVNNDMIVTDNNGAVK...      NaN        []   
376232  MEPGRGGTETVGKFEFSRKDLIGHGAFAVVFKGRHREKHDLEVAVK...      NaN        []   

       mutants  target_id Protein_Type  
0           []  Q9Y251_WT           WT  
1           []  P07872_WT           WT  
2           []  P05177_WT           WT  
3           []  P08684_WT           WT  
4           []  P11712_WT           WT  
...        ...        ...          ...  
376228      []  Q9NRM7_WT           WT  
376229      []  Q13464_WT           WT  
376230      []  O75116_WT           WT  
376231      []  P37173_WT           WT  
376232      []  O75385_WT           WT  

[376233 rows x 9 columns]

In [20]:
# Check newly annotated assays (not previously annotated on ChEMBL)
filter_positive_annotations(assays_previous_round, 'new')

assay_id                                        description accession  \
3103      226874  Ratio of IC50 value against hNK1 receptor to t...    P25103   
5557      153379  Transcriptional activation activity on human I...    Q07869   
5558      153380  Transcriptional activation activity on human T...    Q07869   
5559      153557  Ratio of transcriptional activation of I272F m...    Q07869   
5560      153560  Ratio of transcriptional activation of T279M m...    Q07869   
...          ...                                                ...       ...   
371084   2113388  Inhibition of human non-phosphorylated ABL1 F3...    P00519   
373669   2122982  Inhibition of MLL1 (unknown origin)-mediated H...    Q03164   
374059   2125019  Activation of human PKM2-C424A expressed in Es...    P14618   
374624   2127601  Binding affinity to human partial length MCL1 ...    Q07820   
375932   2135004  Protac activity at CRBN/AR T878A mutant in hum...    P10275   

                                                 sequence mutation  \
3103    MDNVLPVDSDLSPNISTNTSEPNQFVQPAWQIVLWAAAYTVIVVTS...      NaN   
5557    MVDTESPLCPLSPLEAGDLESPLSEEFLQEMGNIQEISQSIGEDSS...      NaN   
5558    MVDTESPLCPLSPLEAGDLESPLSEEFLQEMGNIQEISQSIGEDSS...      NaN   
5559    MVDTESPLCPLSPLEAGDLESPLSEEFLQEMGNIQEISQSIGEDSS...      NaN   
5560    MVDTESPLCPLSPLEAGDLESPLSEEFLQEMGNIQEISQSIGEDSS...      NaN   
...                                                   ...      ...   
371084  MLEICLKLVGCKSKKGLSSSSSCYLEEALQRPVASDFEPQGLSEAA...      NaN   
373669  MAHSCRWRFPARPGTTGGGGGGGRRGLGGAPRQRVPALLLPPGPPV...      NaN   
374059  MSKPHSEAGTAFIQTQQLHAAMADTFLEHMCRLDIDSPPITARNTG...      NaN   
374624  MFGLKRNAVIGLNLYCGGAGLGAGSGGATRPGGRLLATEKEASARR...      NaN   
375932  MEVQLGLGRVYPRPPSKTYRGAFQNLFQSVREVIQNPGPRHPEAAS...      NaN   

           aa_change  mutants     target_id Protein_Type  
3103         [Q165A]  [Q165A]  P25103_Q165A        Q165A  
5557         [I272F]  [I272F]  Q07869_I272F        I272F  
5558         [T279M]  [T279M]  Q07869_T279M        T279M  
5559         [I272F]  [I272F]  Q07869_I272F        I272F  
5560         [T279M]  [T279M]  Q07869_T279M        T279M  
...              ...      ...           ...          ...  
371084       [F317L]  [F317L]  P00519_F317L        F317L  
373669         [H3K]    [H3K]    Q03164_H3K          H3K  
374059       [C424A]  [C424A]  P14618_C424A        C424A  
374624  [D172N, L2S]  [D172N]  Q07820_D172N        D172N  
375932       [T878A]  [T878A]  P10275_T878A        T878A  

[651 rows x 9 columns]

In [21]:
# Check rescued assays (annotated as 'UNDEFINED MUTATION' in ChEMBL)
filter_positive_annotations(assays_previous_round, 'rescued')

assay_id                                        description accession  \
225249   1433633  Potentiation of human CFTR F508del/G551D mutan...    P13569   
227270   1442036  Inhibition of KIT V559/T670I mutant (unknown o...    P10721   
228081   1435798  Inhibition of wild type HIV1 reverse transcrip...    Q72547   
228390   1440950  Inhibition of EGFR T790M/del746 to 750 mutant ...    P00533   
228616   1441757  Corrector activity at human bronchial epitheli...    P13569   
...          ...                                                ...       ...   
374540   2127220  Binding affinity to 6His-FLAG-Tev-BRDT (1 to 3...    Q58F21   
374554   2127280  Inhibition of 6His-FLAG-Tev-BRDT (1 to 397 res...    Q58F21   
374556   2127282  Inhibition of 6His-Thr-BRD3 (1 to 435 residues...    Q15059   
374557   2127283  Inhibition of 6His-Thr-BRD2 (1 to 473 residues...    P25440   
375853   2134720  Inhibition of EGFR d747-749/A750P mutant (unkn...    P00533   

                                                 sequence            mutation  \
225249  MQRSPLEKASVVSKLFFSWTRPILRKGYRQRLELSDIYQIPSVDSA...  UNDEFINED MUTATION   
227270  MRGARGAWDFLCVLLLLLRVQTGSSQPSVSPGEPSPPSIHPGKSDL...  UNDEFINED MUTATION   
228081  PISPIETVPVKLKPGMDGPKVKQWPLTEEKIKALVEICTEMEKEGK...  UNDEFINED MUTATION   
228390  MRPSGTAGAALLALLAALCPASRALEEKKVCQGTSNKLTQLGTFED...  UNDEFINED MUTATION   
228616  MQRSPLEKASVVSKLFFSWTRPILRKGYRQRLELSDIYQIPSVDSA...  UNDEFINED MUTATION   
...                                                   ...                 ...   
374540  MSLPSRQTAIIVNPPPPEYINTKKNGRLTNQLQYLQKVVLKDLWKH...  UNDEFINED MUTATION   
374554  MSLPSRQTAIIVNPPPPEYINTKKNGRLTNQLQYLQKVVLKDLWKH...  UNDEFINED MUTATION   
374556  MSTATTVAPAGIPATPGPVNPPPPEVSNPSKPGRKTNQLQYMQNVV...  UNDEFINED MUTATION   
374557  MLQNVTPHNKLPGEGNAGLLGLGPEAAAPGKRIRKPSLLYEGFESP...  UNDEFINED MUTATION   
375853  MRPSGTAGAALLALLAALCPASRALEEKKVCQGTSNKLTQLGTFED...  UNDEFINED MUTATION   

             aa_change         mutants           target_id Protein_Type  
225249         [G551D]         [G551D]        P13569_G551D        G551D  
227270         [T670I]         [T670I]        P10721_T670I        T670I  
228081  [Y181I, Y181C]  [Y181C, Y181I]  Q72547_Y181C_Y181I  Y181C;Y181I  
228390         [T790M]         [T790M]        P00533_T790M        T790M  
228616         [G551D]         [G551D]        P13569_G551D        G551D  
...                ...             ...                 ...          ...  
374540   [Y309A, Y66A]   [Y66A, Y309A]   Q58F21_Y66A_Y309A   Y66A;Y309A  
374554   [Y309A, Y66A]   [Y66A, Y309A]   Q58F21_Y66A_Y309A   Y66A;Y309A  
374556   [Y348A, Y73A]   [Y73A, Y348A]   Q15059_Y73A_Y348A   Y73A;Y348A  
374557  [Y386A, Y113A]  [Y113A, Y386A]  P25440_Y113A_Y386A  Y113A;Y386A  
375853         [A750P]         [A750P]        P00533_A750P        A750P  

[182 rows x 9 columns]

In [47]:
# Check missed annotations by automatic annotation 
filter_negative_annotations(assays_previous_round)

assay_id                                        description accession  \
88483     317562  Binding affinity for mutant T877A Androgen rec...    P10275   
94476     322454  Inhibitory concentration against IL-3 independ...    P10721   
105052    426971  Ratio of Ki to Km for HCV NS5B RNA polymerase ...    Q8JXU8   
105060    426978  Activity of HCV NS5B RNA polymerase S282T asse...    Q8JXU8   
107598    445385  Agonist activity at GluR1 I687A mutant express...    P42261   
...          ...                                                ...       ...   
373946   2124228  Stabilization of TTR V3OM mutant (unknown orig...    P02766   
373947   2124230  Stabilization of TTR V3OM mutant (unknown orig...    P02766   
373949   2124234  Binding affinity to TTR V30M mutant (unknown o...    P02766   
375931   2135003  Protac activity at CRBN/AR T878A mutant in hum...    Q96SW2   
375937   2135067  Protac activity at CRBN/AR T878A mutant in hum...    Q96SW2   

                                                 sequence mutation  \
88483   MEVQLGLGRVYPRPPSKTYRGAFQNLFQSVREVIQNPGPRHPEAAS...    T878A   
94476   MRGARGAWDFLCVLLLLLRVQTGSSQPSVSPGEPSPPSIHPGKSDL...    D816V   
105052  RTEEAIYQCCDLDPQARVAIRSLTERLYVGGPLTNSRGENCGYRRX...    S282T   
105060  RTEEAIYQCCDLDPQARVAIRSLTERLYVGGPLTNSRGENCGYRRX...    S282T   
107598  MQHIFAFFCTGFLGAVVGANFPNNIQIGGLFPNQQSQEHAAFRFAL...    I705A   
...                                                   ...      ...   
373946  MASHRLLLLCLAGLVFVSEAGPTGTGESKCPLMVKVLDAVRGSPAI...     V50M   
373947  MASHRLLLLCLAGLVFVSEAGPTGTGESKCPLMVKVLDAVRGSPAI...     V50M   
373949  MASHRLLLLCLAGLVFVSEAGPTGTGESKCPLMVKVLDAVRGSPAI...     V50M   
375931  MAGEGDQQDAAHNMGNHLPLLPAESEEEDEMEVEDQDSKEAKKPNI...    T878A   
375937  MAGEGDQQDAAHNMGNHLPLLPAESEEEDEMEVEDQDSKEAKKPNI...    T878A   

           aa_change mutants  target_id Protein_Type  
88483        [T877A]      []  P10275_WT           WT  
94476        [D816V]      []  P10721_WT           WT  
105052  [S5B, S282T]      []  Q8JXU8_WT           WT  
105060  [S5B, S282T]      []  Q8JXU8_WT           WT  
107598       [I687A]      []  P42261_WT           WT  
...              ...     ...        ...          ...  
373946         [V3O]      []  P02766_WT           WT  
373947         [V3O]      []  P02766_WT           WT  
373949        [V30M]      []  P02766_WT           WT  
375931       [T878A]      []  Q96SW2_WT           WT  
375937       [T878A]      []  Q96SW2_WT           WT  

[1600 rows x 9 columns]

##### Check automatic annotations that were not present in ChEMBL and manually flagged as false positives

The annotation of false positives was done by manually checking the new annotations and "rescued" UNDEFINED
MUTATION assay-accession pairs (651 and 182 in round 1). The assays descriptions were checked against the annotated
mutations. Assay-accession pairs that were considered false positives were extracted in 
anotation_analysis_dir\new_annotations_incorrect.xlsx and given a
"reason" flag that described that reason why the annotated mutation was likely given. This should help categorize the
 false positives and give ideas to improve the annotation pipeline. The different reasons were further grouped into
 more general groups in "group_reason" flag.

In [14]:
false_positives = read_manual_positive_annotations(chembl_version, previous_annotation_round)
false_positives

File with manual annotations for false positives found. Continuing...


assay_id accession mutants            reason  N      group_reason
0      208643    P25103   Sar9M             assay  1             assay
1      801913    A4HSF7    A67I           extract  1             assay
2      801916    A4HSF7    A67I           extract  1             assay
3      801918    A4HSF7    A67I           extract  1             assay
4     1791091    P11678   [H2O]              H2O2  1             assay
..        ...       ...     ...               ... ..               ...
296   1881081    P10721   V559D  missing mutation  1  missing mutation
297   1887622    P10721   V559D  missing mutation  1  missing mutation
298   1888099    P10721   V559D  missing mutation  1  missing mutation
299   1889648    P10721   V559D  missing mutation  1  missing mutation
300   1886716    P58335    R40C  missing mutation  1  missing mutation

[301 rows x 6 columns]

##### Check ChEMBL annotations that were missed in the automatic annotation and their likely reasons for rejection

The annotation of false negatives was done by classifying all rejected ChEMBL mutations into different groups (i.e.
"rejection_flag") defined from an initial visual inspection and talks to the variant annotation team in ChEMBL.
Further relevance/confidence of the different categories were given by further visual inspection.

In [23]:
# Read negative assays with false negative rejection flags given by automatic classification
false_negatives = classify_negative_annotations(chembl_version, previous_annotation_round)
false_negatives

assay_id                                      description_x accession  \
0       317562  Binding affinity for mutant T877A Androgen rec...    P10275   
1       322454  Inhibitory concentration against IL-3 independ...    P10721   
2       426971  Ratio of Ki to Km for HCV NS5B RNA polymerase ...    Q8JXU8   
3       426978  Activity of HCV NS5B RNA polymerase S282T asse...    Q8JXU8   
4       445385  Agonist activity at GluR1 I687A mutant express...    P42261   
...        ...                                                ...       ...   
1595   2124228  Stabilization of TTR V3OM mutant (unknown orig...    P02766   
1596   2124230  Stabilization of TTR V3OM mutant (unknown orig...    P02766   
1597   2124234  Binding affinity to TTR V30M mutant (unknown o...    P02766   
1598   2135003  Protac activity at CRBN/AR T878A mutant in hum...    Q96SW2   
1599   2135067  Protac activity at CRBN/AR T878A mutant in hum...    Q96SW2   

                                             sequence_x mutation  \
0     MEVQLGLGRVYPRPPSKTYRGAFQNLFQSVREVIQNPGPRHPEAAS...    T878A   
1     MRGARGAWDFLCVLLLLLRVQTGSSQPSVSPGEPSPPSIHPGKSDL...    D816V   
2     RTEEAIYQCCDLDPQARVAIRSLTERLYVGGPLTNSRGENCGYRRX...    S282T   
3     RTEEAIYQCCDLDPQARVAIRSLTERLYVGGPLTNSRGENCGYRRX...    S282T   
4     MQHIFAFFCTGFLGAVVGANFPNNIQIGGLFPNQQSQEHAAFRFAL...    I705A   
...                                                 ...      ...   
1595  MASHRLLLLCLAGLVFVSEAGPTGTGESKCPLMVKVLDAVRGSPAI...     V50M   
1596  MASHRLLLLCLAGLVFVSEAGPTGTGESKCPLMVKVLDAVRGSPAI...     V50M   
1597  MASHRLLLLCLAGLVFVSEAGPTGTGESKCPLMVKVLDAVRGSPAI...     V50M   
1598  MAGEGDQQDAAHNMGNHLPLLPAESEEEDEMEVEDQDSKEAKKPNI...    T878A   
1599  MAGEGDQQDAAHNMGNHLPLLPAESEEEDEMEVEDQDSKEAKKPNI...    T878A   

             aa_change mutants  target_id Protein_Type  doc_id  ...  \
0            ['T877A']      []  P10275_WT           WT   20674  ...   
1            ['D816V']      []  P10721_WT           WT   30197  ...   
2     ['S5B', 'S282T']      []  Q8JXU8_WT           WT   35099  ...   
3     ['S5B', 'S282T']      []  Q8JXU8_WT           WT   35099  ...   
4            ['I687A']      []  P42261_WT           WT   36637  ...   
...                ...     ...        ...          ...     ...  ...   
1595           ['V3O']      []  P02766_WT           WT  120955  ...   
1596           ['V3O']      []  P02766_WT           WT  120955  ...   
1597          ['V30M']      []  P02766_WT           WT  120955  ...   
1598         ['T878A']      []  Q96SW2_WT           WT  121349  ...   
1599         ['T878A']      []  Q96SW2_WT           WT  121349  ...   

     component_id.1 component_type  \
0               187        PROTEIN   
1               255        PROTEIN   
2              3671        PROTEIN   
3              3671        PROTEIN   
4              4996        PROTEIN   
...             ...            ...   
1595           1517        PROTEIN   
1596           1517        PROTEIN   
1597           1517        PROTEIN   
1598          16927        PROTEIN   
1599          16927        PROTEIN   

                                             sequence_y  \
0     MEVQLGLGRVYPRPPSKTYRGAFQNLFQSVREVIQNPGPRHPEAAS...   
1     MRGARGAWDFLCVLLLLLRVQTGSSQPSVSPGEPSPPSIHPGKSDL...   
2     RTEEAIYQCCDLDPQARVAIRSLTERLYVGGPLTNSRGENCGYRRX...   
3     RTEEAIYQCCDLDPQARVAIRSLTERLYVGGPLTNSRGENCGYRRX...   
4     MQHIFAFFCTGFLGAVVGANFPNNIQIGGLFPNQQSQEHAAFRFAL...   
...                                                 ...   
1595  MASHRLLLLCLAGLVFVSEAGPTGTGESKCPLMVKVLDAVRGSPAI...   
1596  MASHRLLLLCLAGLVFVSEAGPTGTGESKCPLMVKVLDAVRGSPAI...   
1597  MASHRLLLLCLAGLVFVSEAGPTGTGESKCPLMVKVLDAVRGSPAI...   
1598  MAGEGDQQDAAHNMGNHLPLLPAESEEEDEMEVEDQDSKEAKKPNI...   
1599  MAGEGDQQDAAHNMGNHLPLLPAESEEEDEMEVEDQDSKEAKKPNI...   

                       sequence_md5sum  \
0     0e8126a2b4e057dff4e5e448802e64bf   
1     f753f2b2d9752a30646c83e8630d1df8   
2     3f5012b8af293164261075aeb623fec5   
3     3f5012b8af293164261075aeb623fec5   
4     18757

In [24]:
# Check assays in specific rejection flags
false_negatives[false_negatives['rejection_flag'] == 'original_not_valid']

assay_id                                      description_x accession  \
1       322454  Inhibitory concentration against IL-3 independ...    P10721   
2       426971  Ratio of Ki to Km for HCV NS5B RNA polymerase ...    Q8JXU8   
3       426978  Activity of HCV NS5B RNA polymerase S282T asse...    Q8JXU8   
15      473343  Binding affinity to NR2B receptor S645A/i641A ...    Q13224   
19      534387  Inhibition of cross-linking of human Pgp TM3T1...    P08183   
...        ...                                                ...       ...   
1586   2111448  Inhibition of p110alpha E542K mutant/p85alpha ...    P27986   
1588   2117080  Protac activity at CRBN/EGFR L858R/T790M mutan...    Q96SW2   
1589   2117102  Protac activity at CRBN/EGFR L858R/T790M mutan...    Q96SW2   
1598   2135003  Protac activity at CRBN/AR T878A mutant in hum...    Q96SW2   
1599   2135067  Protac activity at CRBN/AR T878A mutant in hum...    Q96SW2   

                                             sequence_x           mutation  \
1     MRGARGAWDFLCVLLLLLRVQTGSSQPSVSPGEPSPPSIHPGKSDL...              D816V   
2     RTEEAIYQCCDLDPQARVAIRSLTERLYVGGPLTNSRGENCGYRRX...              S282T   
3     RTEEAIYQCCDLDPQARVAIRSLTERLYVGGPLTNSRGENCGYRRX...              S282T   
15    MKPRAECCSPKFWLVLAVLAVSGSRARSQKSPPSIGIAVILVGTSD...        S645A,I641A   
19    MDLEGDRNGGAKKKNFFKLNNKSEKDKKEKKPTVSVFSMFRYSNWL...  T199R,L339C,F728C   
...                                                 ...                ...   
1586  MSAEGYQYRALYDYKKEREEDIDLHLGDILTVNKGSLVALGFSDGQ...              E542K   
1588  MAGEGDQQDAAHNMGNHLPLLPAESEEEDEMEVEDQDSKEAKKPNI...        L858R,T790M   
1589  MAGEGDQQDAAHNMGNHLPLLPAESEEEDEMEVEDQDSKEAKKPNI...        L858R,T790M   
1598  MAGEGDQQDAAHNMGNHLPLLPAESEEEDEMEVEDQDSKEAKKPNI...              T878A   
1599  MAGEGDQQDAAHNMGNHLPLLPAESEEEDEMEVEDQDSKEAKKPNI...              T878A   

                      aa_change             mutants           target_id  \
1                     ['D816V']                  []           P10721_WT   
2              ['S5B', 'S282T']                  []           Q8JXU8_WT   
3              ['S5B', 'S282T']                  []           Q8JXU8_WT   
15             ['R2B', 'S645A']           ['S645A']        Q13224_S645A   
19    ['M3T', 'L339C', 'F728C']  ['L339C', 'F728C']  P08183_L339C_F728C   
...                         ...                 ...                 ...   
1586                  ['E542K']                  []           P27986_WT   
1588         ['L858R', 'T790M']                  []           Q96SW2_WT   
1589         ['L858R', 'T790M']                  []           Q96SW2_WT   
1598                  ['T878A']                  []           Q96SW2_WT   
1599                  ['T878A']                  []           Q96SW2_WT   

     Protein_Type  doc_id  ... component_id.1 component_type  \
1              WT   30197  ...            255        PROTEIN   
2              WT   35099  ...           3671        PROTEIN   
3              WT   35099  ...           3671        PROTEIN   
15          S645A   39129  ...             59        PROTEIN   
19    L339C;F728C   42551  ...           2619        PROTEIN   
...           ...     ...  ...            ...            ...   
1586           WT  120456  ...            846        PROTEIN   
1588           WT  120695  ...          16927        PROTEIN   
1589           WT  120695  ...          16927        PROTEIN   
1598           WT  121349  ...          16927        PROTEIN   
1599           WT  121349  ...          16927        PROTEIN   

                                             sequence_y  \
1     MRGARGAWDFLCVLLLLLRVQTGSSQPSVSPGEPSPPSIHPGKSDL...   
2     RTEEAIYQCCDLDPQARVAIRSLTERLYVGGPLTNSRGENCGYRRX...   
3     RTEEAIYQCCDLDPQARVAIRSLTERLYVGGPLTNSRGENCGYRRX...   
15    MKPRAECCSPKFWLVLAVLAVSGSRARSQKSPPSIGIAVILVGTSD...   
19    MDLEGDRNGGAKKKNFFKLNNKSEKDKKEKKPTVSVFSMFRYSNWL...   
...                                                 ...   
1586  MSAEGYQYRALYDYKKEREEDIDLHLGDILT

In [25]:
# Check how many rejected fall under each rejection category
false_negatives.groupby(['rejection_flag'])['assay_id'].count()

rejection_flag
no_extraction                      175
original_deletion                   77
original_exception_Autocuration    168
original_not_valid                 248
original_shift_exception           323
protein_family                     609
Name: assay_id, dtype: int64

#### Re-annotate assays based on the flags given for false positives and negatives

In [26]:
new_round_assays = manual_reannotation(chembl_version=chembl_version,
                    annotation_round=annotation_round,
                    correct_false_positives=True,
                    correct_false_negatives=True)
new_round_assays

assay_id                                        description accession  \
0          83907  In vivo inhibitory activity against human Hepa...    Q9Y251   
1         154606  Inhibitory activity against Palmitoyl-CoA oxid...    P07872   
2          51352  Inhibition of cytochrome P450 1A2 of isolated ...    P05177   
3          51895  Inhibition of cytochrome P450 3A4 of isolated ...    P08684   
4          51528  Inhibition of cytochrome P450 2C9 of isolated ...    P11712   
...          ...                                                ...       ...   
376228   2136225              KinomeScan assay: inhibition of LATS2    Q9NRM7   
376229   2136385              KinomeScan assay: inhibition of ROCK1    Q13464   
376230   2136386              KinomeScan assay: inhibition of ROCK2    O75116   
376231   2136430             KinomeScan assay: inhibition of TGFBR2    P37173   
376232   2136450               KinomeScan assay: inhibition of ULK1    O75385   

                                                 sequence mutation aa_change  \
0       MLLRSKPALPPPLMLLLLGPLGPLSPGALPRPAQAQDVVDLDFFTQ...      NaN        []   
1       MNPDLRKERASATFNPELITHILDGSPENTRRRREIENLILNDPDF...      NaN        []   
2       MALSQSVPFSATELLLASAIFCLVFWVLKGLRPRVPKGLKSPPEPW...      NaN        []   
3       MALIPDLAMETWLLLAVSLVLLYLYGTHSHGLFKKLGIPGPTPLPF...      NaN        []   
4       MDSLVVLVLCLSCLLLLSLWRQSSGRGKLPPGPTPLPVIGNILQIG...      NaN        []   
...                                                   ...      ...       ...   
376228  MRPKTFPATTYSGNSRQRLQEIREGLKQPSKSSVQGLPAGPNSDTS...      NaN        []   
376229  MSTGDSFETRFEKMDNLLRDPKSEVNSDCLLDGLDALVYDLDFPAL...      NaN        []   
376230  MSRPPPTGKMPGAPETAPGDGAGASRQRKLEALIRDPRSPINVESL...      NaN        []   
376231  MGRGLLRGLWPLHIVLWTRIASTIPPHVQKSVNNDMIVTDNNGAVK...      NaN        []   
376232  MEPGRGGTETVGKFEFSRKDLIGHGAFAVVFKGRHREKHDLEVAVK...      NaN        []   

       mutants  target_id Protein_Type  
0           []  Q9Y251_WT           WT  
1           []  P07872_WT           WT  
2           []  P05177_WT           WT  
3           []  P08684_WT           WT  
4           []  P11712_WT           WT  
...        ...        ...          ...  
376228      []  Q9NRM7_WT           WT  
376229      []  Q13464_WT           WT  
376230      []  O75116_WT           WT  
376231      []  P37173_WT           WT  
376232      []  O75385_WT           WT  

[376233 rows x 9 columns]

In [27]:
# Check how many assays have mutant annotations (including undefined mutations)
new_round_assays[~new_round_assays['target_id'].str.contains('WT')]

assay_id                                        description accession  \
2008      164502  Inhibitory activity against HIV-1 reverse tran...    Q72547   
2009      164500  Inhibitory activity against HIV-1 reverse tran...    Q72547   
2163      164503  Substrate Activity against HIV-1 reverse trans...    Q72547   
2164      164501  Substrate Activity against HIV-1 reverse trans...    Q72547   
3102      205569  Binding affinity against Gln165Ala mutant type...    P25103   
...          ...                                                ...       ...   
375937   2135067  Protac activity at CRBN/AR T878A mutant in hum...    Q96SW2   
375967   2135214   Inhibition of EGFR L858R mutant (unknown origin)    P00533   
376193   2135936  Inhibition of GDP-loaded His-tagged KRAS G12C ...    P01116   
376194   2135937  Inhibition of KRAS G12C mutant in human NCI-H3...    P01116   
376195   2135940  Displacement of Cy5-labelled tracer from bioti...    P01116   

                                                 sequence mutation  aa_change  \
2008    PISPIETVPVKLKPGMDGPKVKQWPLTEEKIKALVEICTEMEKEGK...    M184V  ['M184V']   
2009    PISPIETVPVKLKPGMDGPKVKQWPLTEEKIKALVEICTEMEKEGK...    M184I  ['M184I']   
2163    PISPIETVPVKLKPGMDGPKVKQWPLTEEKIKALVEICTEMEKEGK...    M184V  ['M184V']   
2164    PISPIETVPVKLKPGMDGPKVKQWPLTEEKIKALVEICTEMEKEGK...    M184I  ['M184I']   
3102    MDNVLPVDSDLSPNISTNTSEPNQFVQPAWQIVLWAAAYTVIVVTS...    Q165A  ['Q165A']   
...                                                   ...      ...        ...   
375937  MAGEGDQQDAAHNMGNHLPLLPAESEEEDEMEVEDQDSKEAKKPNI...    T878A  ['T878A']   
375967  MRPSGTAGAALLALLAALCPASRALEEKKVCQGTSNKLTQLGTFED...    L858R  ['L858R']   
376193  MTEYKLVVVGAGGVGKSALTIQLIQNHFVDEYDPTIEDSYRKQVVI...     G12C   ['G12C']   
376194  MTEYKLVVVGAGGVGKSALTIQLIQNHFVDEYDPTIEDSYRKQVVI...     G12C   ['G12C']   
376195  MTEYKLVVVGAGGVGKSALTIQLIQNHFVDEYDPTIEDSYRKQVVI...     G12D   ['G12D']   

          mutants      target_id Protein_Type  
2008    ['M184V']   Q72547_M184V        M184V  
2009    ['M184I']   Q72547_M184I        M184I  
2163    ['M184V']   Q72547_M184V        M184V  
2164    ['M184I']   Q72547_M184I        M184I  
3102    ['Q165A']   P25103_Q165A        Q165A  
...           ...            ...          ...  
375937         []  Q96SW2_MUTANT           WT  
375967  ['L858R']   P00533_L858R        L858R  
376193   ['G12C']    P01116_G12C         G12C  
376194   ['G12C']    P01116_G12C         G12C  
376195   ['G12D']    P01116_G12D         G12D  

[9229 rows x 9 columns]

In [33]:
# Check how many assays now have new annotations compared to ChEMBL after reannotation
# (undefined mutations are not considered positive annotations)
filter_positive_annotations(new_round_assays, 'new')

assay_id                                        description accession  \
3103      226874  Ratio of IC50 value against hNK1 receptor to t...    P25103   
5557      153379  Transcriptional activation activity on human I...    Q07869   
5558      153380  Transcriptional activation activity on human T...    Q07869   
5559      153557  Ratio of transcriptional activation of I272F m...    Q07869   
5560      153560  Ratio of transcriptional activation of T279M m...    Q07869   
...          ...                                                ...       ...   
367794   2098264  Inhibition of human GCN2 S808G mutant kinase d...    Q9P2K8   
371084   2113388  Inhibition of human non-phosphorylated ABL1 F3...    P00519   
374059   2125019  Activation of human PKM2-C424A expressed in Es...    P14618   
374624   2127601  Binding affinity to human partial length MCL1 ...    Q07820   
375932   2135004  Protac activity at CRBN/AR T878A mutant in hum...    P10275   

                                                 sequence mutation  \
3103    MDNVLPVDSDLSPNISTNTSEPNQFVQPAWQIVLWAAAYTVIVVTS...      NaN   
5557    MVDTESPLCPLSPLEAGDLESPLSEEFLQEMGNIQEISQSIGEDSS...      NaN   
5558    MVDTESPLCPLSPLEAGDLESPLSEEFLQEMGNIQEISQSIGEDSS...      NaN   
5559    MVDTESPLCPLSPLEAGDLESPLSEEFLQEMGNIQEISQSIGEDSS...      NaN   
5560    MVDTESPLCPLSPLEAGDLESPLSEEFLQEMGNIQEISQSIGEDSS...      NaN   
...                                                   ...      ...   
367794  MAGGRGAPGRGRDEPPESYPQRQDHELQALEAIYGADFQDLRPDAC...      NaN   
371084  MLEICLKLVGCKSKKGLSSSSSCYLEEALQRPVASDFEPQGLSEAA...      NaN   
374059  MSKPHSEAGTAFIQTQQLHAAMADTFLEHMCRLDIDSPPITARNTG...      NaN   
374624  MFGLKRNAVIGLNLYCGGAGLGAGSGGATRPGGRLLATEKEASARR...      NaN   
375932  MEVQLGLGRVYPRPPSKTYRGAFQNLFQSVREVIQNPGPRHPEAAS...      NaN   

               aa_change    mutants     target_id Protein_Type  
3103           ['Q165A']  ['Q165A']  P25103_Q165A        Q165A  
5557           ['I272F']  ['I272F']  Q07869_I272F        I272F  
5558           ['T279M']  ['T279M']  Q07869_T279M        T279M  
5559           ['I272F']  ['I272F']  Q07869_I272F        I272F  
5560           ['T279M']  ['T279M']  Q07869_T279M        T279M  
...                  ...        ...           ...          ...  
367794         ['S808G']  ['S808G']  Q9P2K8_S808G        S808G  
371084         ['F317L']  ['F317L']  P00519_F317L        F317L  
374059         ['C424A']  ['C424A']  P14618_C424A        C424A  
374624  ['D172N', 'L2S']  ['D172N']  Q07820_D172N        D172N  
375932         ['T878A']  ['T878A']  P10275_T878A        T878A  

[538 rows x 9 columns]

In [34]:
# Check how many assay now have rescued annotations compared to ChEMBL after reannotation
# (undefined mutations are not considered positive annotations)
filter_positive_annotations(new_round_assays, 'rescued')

assay_id                                        description accession  \
225249   1433633  Potentiation of human CFTR F508del/G551D mutan...    P13569   
228081   1435798  Inhibition of wild type HIV1 reverse transcrip...    Q72547   
228390   1440950  Inhibition of EGFR T790M/del746 to 750 mutant ...    P00533   
228616   1441757  Corrector activity at human bronchial epitheli...    P13569   
231419   1472456  Inhibition of FLT3 ITD D835V mutant (unknown o...    P36888   
...          ...                                                ...       ...   
372157   2114787  Inhibition of FLT3 ITD/F691L double mutant (un...    P36888   
372171   2114825  Inhibition of FLT3 ITD/D835Y double mutant (un...    P36888   
372172   2114826  Inhibition of FLT3 ITD/D835V double mutant (un...    P36888   
373444   2121848  Inhibition of EGFR del19/T790M/C797S triple mu...    P00533   
375853   2134720  Inhibition of EGFR d747-749/A750P mutant (unkn...    P00533   

                                                 sequence            mutation  \
225249  MQRSPLEKASVVSKLFFSWTRPILRKGYRQRLELSDIYQIPSVDSA...  UNDEFINED MUTATION   
228081  PISPIETVPVKLKPGMDGPKVKQWPLTEEKIKALVEICTEMEKEGK...  UNDEFINED MUTATION   
228390  MRPSGTAGAALLALLAALCPASRALEEKKVCQGTSNKLTQLGTFED...  UNDEFINED MUTATION   
228616  MQRSPLEKASVVSKLFFSWTRPILRKGYRQRLELSDIYQIPSVDSA...  UNDEFINED MUTATION   
231419  MPALARDGGQLPLLVVFSAMIFGTITNQDLPVIKCVLINHKNNDSS...  UNDEFINED MUTATION   
...                                                   ...                 ...   
372157  MPALARDGGQLPLLVVFSAMIFGTITNQDLPVIKCVLINHKNNDSS...  UNDEFINED MUTATION   
372171  MPALARDGGQLPLLVVFSAMIFGTITNQDLPVIKCVLINHKNNDSS...  UNDEFINED MUTATION   
372172  MPALARDGGQLPLLVVFSAMIFGTITNQDLPVIKCVLINHKNNDSS...  UNDEFINED MUTATION   
373444  MRPSGTAGAALLALLAALCPASRALEEKKVCQGTSNKLTQLGTFED...  UNDEFINED MUTATION   
375853  MRPSGTAGAALLALLAALCPASRALEEKKVCQGTSNKLTQLGTFED...  UNDEFINED MUTATION   

                 aa_change             mutants           target_id  \
225249           ['G551D']           ['G551D']        P13569_G551D   
228081  ['Y181I', 'Y181C']  ['Y181C', 'Y181I']  Q72547_Y181C_Y181I   
228390           ['T790M']           ['T790M']        P00533_T790M   
228616           ['G551D']           ['G551D']        P13569_G551D   
231419           ['D835V']           ['D835V']        P36888_D835V   
...                    ...                 ...                 ...   
372157           ['F691L']           ['F691L']        P36888_F691L   
372171           ['D835Y']           ['D835Y']        P36888_D835Y   
372172           ['D835V']           ['D835V']        P36888_D835V   
373444  ['T790M', 'C797S']  ['T790M', 'C797S']  P00533_T790M_C797S   
375853           ['A750P']           ['A750P']        P00533_A750P   

       Protein_Type  
225249        G551D  
228081  Y181C;Y181I  
228390        T790M  
228616        G551D  
231419        D835V  
...             ...  
372157        F691L  
372171        D835Y  
372172        D835V  
373444  T790M;C797S  
375853        A750P  

[162 rows x 9 columns]

In [35]:
# Check how many annotated assays are annotated with an undefined label
new_round_assays[new_round_assays['target_id'].str.contains('MUTANT')]

assay_id                                        description accession  \
94476     322454  Inhibitory concentration against IL-3 independ...    P10721   
105052    426971  Ratio of Ki to Km for HCV NS5B RNA polymerase ...    Q8JXU8   
105060    426978  Activity of HCV NS5B RNA polymerase S282T asse...    Q8JXU8   
114613    473331  Binding affinity to NR2B receptor M3c T647A mu...    Q13224   
114623    473343  Binding affinity to NR2B receptor S645A/i641A ...    Q13224   
...          ...                                                ...       ...   
374554   2127280  Inhibition of 6His-FLAG-Tev-BRDT (1 to 397 res...    Q58F21   
374556   2127282  Inhibition of 6His-Thr-BRD3 (1 to 435 residues...    Q15059   
374557   2127283  Inhibition of 6His-Thr-BRD2 (1 to 473 residues...    P25440   
375931   2135003  Protac activity at CRBN/AR T878A mutant in hum...    Q96SW2   
375937   2135067  Protac activity at CRBN/AR T878A mutant in hum...    Q96SW2   

                                                 sequence            mutation  \
94476   MRGARGAWDFLCVLLLLLRVQTGSSQPSVSPGEPSPPSIHPGKSDL...               D816V   
105052  RTEEAIYQCCDLDPQARVAIRSLTERLYVGGPLTNSRGENCGYRRX...               S282T   
105060  RTEEAIYQCCDLDPQARVAIRSLTERLYVGGPLTNSRGENCGYRRX...               S282T   
114613  MKPRAECCSPKFWLVLAVLAVSGSRARSQKSPPSIGIAVILVGTSD...               T647T   
114623  MKPRAECCSPKFWLVLAVLAVSGSRARSQKSPPSIGIAVILVGTSD...         S645A,I641A   
...                                                   ...                 ...   
374554  MSLPSRQTAIIVNPPPPEYINTKKNGRLTNQLQYLQKVVLKDLWKH...  UNDEFINED MUTATION   
374556  MSTATTVAPAGIPATPGPVNPPPPEVSNPSKPGRKTNQLQYMQNVV...  UNDEFINED MUTATION   
374557  MLQNVTPHNKLPGEGNAGLLGLGPEAAAPGKRIRKPSLLYEGFESP...  UNDEFINED MUTATION   
375931  MAGEGDQQDAAHNMGNHLPLLPAESEEEDEMEVEDQDSKEAKKPNI...               T878A   
375937  MAGEGDQQDAAHNMGNHLPLLPAESEEEDEMEVEDQDSKEAKKPNI...               T878A   

                 aa_change             mutants      target_id Protein_Type  
94476            ['D816V']                  []  P10721_MUTANT           WT  
105052    ['S5B', 'S282T']                  []  Q8JXU8_MUTANT           WT  
105060    ['S5B', 'S282T']                  []  Q8JXU8_MUTANT           WT  
114613    ['R2B', 'T647A']           ['T647A']  Q13224_MUTANT        T647A  
114623    ['R2B', 'S645A']           ['S645A']  Q13224_MUTANT        S645A  
...                    ...                 ...            ...          ...  
374554   ['Y309A', 'Y66A']   ['Y66A', 'Y309A']  Q58F21_MUTANT   Y66A;Y309A  
374556   ['Y348A', 'Y73A']   ['Y73A', 'Y348A']  Q15059_MUTANT   Y73A;Y348A  
374557  ['Y386A', 'Y113A']  ['Y113A', 'Y386A']  P25440_MUTANT  Y113A;Y386A  
375931           ['T878A']                  []  Q96SW2_MUTANT           WT  
375937           ['T878A']                  []  Q96SW2_MUTANT           WT  

[447 rows x 9 columns]

In [46]:
# Check how many ChEMBL-annotated mutants are still fully rejected 
# (undefined mutations are not considered negative annotations)
filter_negative_annotations(new_round_assays,undefined_mutations=False)

assay_id                                        description accession  \
143911    624186  Inhibition of EGFR Leu858Arg and Thr790Met mut...    P04626   
143913    624186  Inhibition of EGFR Leu858Arg and Thr790Met mut...    Q15303   
143914    624186  Inhibition of EGFR Leu858Arg and Thr790Met mut...    P21860   
143915    624185  Inhibition of EGFR Leu858Arg mutant by HTRF assay    P04626   
143917    624185  Inhibition of EGFR Leu858Arg mutant by HTRF assay    Q15303   
...          ...                                                ...       ...   
364470   2086921  Synergistic potentiator activity at CFTR F508d...    P13569   
364471   2086922  Synergistic potentiator activity at CFTR F508d...    P13569   
365662   2094601  Potentiation of CFTR F508del mutant (unknown o...    P13569   
365666   2094620  Potentiation of CFTR F508del mutant in human H...    P13569   
370751   2111684  Binding affinity to sensorchip-immobilized hum...    P13569   

                                                 sequence     mutation  \
143911  MELAALCRWGLLLALLPPGAASTQVCTGTDMKLRLPASPETHLDML...  L858R,T790M   
143913  MKPATGLWVWVSLLVAAGTVQPSDSQSVCAGTENKLSSLSDLEQQY...  L858R,T790M   
143914  MRANDALQVLGLLFSLARGSEVGNSQAVCPGTLNGLSVTGDAENQY...  L858R,T790M   
143915  MELAALCRWGLLLALLPPGAASTQVCTGTDMKLRLPASPETHLDML...        L858R   
143917  MKPATGLWVWVSLLVAAGTVQPSDSQSVCAGTENKLSSLSDLEQQY...        L858R   
...                                                   ...          ...   
364470  MQRSPLEKASVVSKLFFSWTRPILRKGYRQRLELSDIYQIPSVDSA...      F508del   
364471  MQRSPLEKASVVSKLFFSWTRPILRKGYRQRLELSDIYQIPSVDSA...      F508del   
365662  MQRSPLEKASVVSKLFFSWTRPILRKGYRQRLELSDIYQIPSVDSA...      F508del   
365666  MQRSPLEKASVVSKLFFSWTRPILRKGYRQRLELSDIYQIPSVDSA...      F508del   
370751  MQRSPLEKASVVSKLFFSWTRPILRKGYRQRLELSDIYQIPSVDSA...      F508del   

                 aa_change    mutants     target_id Protein_Type  
143911  ['L858R', 'T790M']         []     P04626_WT           WT  
143913  ['L858R', 'T790M']         []     Q15303_WT           WT  
143914  ['L858R', 'T790M']  ['L858R']  P21860_L858R        L858R  
143915           ['L858R']         []     P04626_WT           WT  
143917           ['L858R']         []     Q15303_WT           WT  
...                    ...        ...           ...          ...  
364470                  []         []     P13569_WT           WT  
364471                  []         []     P13569_WT           WT  
365662                  []         []     P13569_WT           WT  
365666                  []         []     P13569_WT           WT  
370751                  []         []     P13569_WT           WT  

[686 rows x 9 columns]

#### Re-annotate bioactivity data for new round based on manual curation

In [48]:
chembl_dataset_new_round = chembl_annotation(chembl_version=chembl_version, annotation_round=annotation_round)
chembl_dataset_new_round

C:\Users\gorostiolam\AppData\Local\Continuum\anaconda3\envs\mutants-in-pcm\lib\site-packages\ipykernel_launcher.py:1: DtypeWarning: Columns (12) have mixed types.Specify dtype option on import or set low_memory=False.
  """Entry point for launching an IPython kernel.


target_id      chembl_id  \
0        A0A024AXB9_WT  CHEMBL1230673   
1        A0A024AXB9_WT  CHEMBL1332616   
2        A0A024AXB9_WT      CHEMBL151   
3        A0A024AXB9_WT     CHEMBL1535   
4        A0A024AXB9_WT      CHEMBL164   
...                ...            ...   
1870743         nan_WT      CHEMBL529   
1870744         nan_WT      CHEMBL532   
1870745         nan_WT   CHEMBL553222   
1870746         nan_WT   CHEMBL557931   
1870747         nan_WT      CHEMBL954   

                                                  assay_id  \
0                                                [1986298]   
1                                       [1986298, 1986301]   
2                                                [1855280]   
3                                                [1986298]   
4                              [1855280, 1855317, 1855317]   
...                                                    ...   
1870743                           [582849, 582849, 582849]   
1870744  [576183, 576183, 576183, 576182, 576182, 57618...   
1870745                           [582849, 582849, 582849]   
1870746                           [582849, 582849, 582849]   
1870747                                          [1854324]   

                                               activity_id   accession  \
0                                               [20667385]  A0A024AXB9   
1                                     [20667410, 20667411]  A0A024AXB9   
2                                               [19053938]  A0A024AXB9   
3                                               [20667415]  A0A024AXB9   
4                           [19053935, 19053936, 19053937]  A0A024AXB9   
...                                                    ...         ...   
1870743                        [2702686, 2702686, 2702686]         NaN   
1870744  [2663798, 2663798, 2663798, 2663799, 2663799, ...         NaN   
1870745                        [2702683, 2702683, 2702683]         NaN   
1870746                        [2702682, 2702682, 2702682]         NaN   
1870747                                         [19051843]         NaN   

                                             pchembl_value  \
0                                                   [6.37]   
1                                             [5.77, 6.22]   
2                                                   [4.28]   
3                                                   [4.04]   
4                                        [4.66, 5.89, 6.1]   
...                                                    ...   
1870743                                 [6.52, 6.52, 6.52]   
1870744  [7.54, 7.54, 7.54, 7.96, 7.96, 7.96, 5.72, 5.7...   
1870745                                    [7.0, 7.0, 7.0]   
1870746                                 [5.89, 5.89, 5.89]   
1870747                                             [5.43]   

                                          activity_comment  \
0                                                    [nan]   
1                                               [nan, nan]   
2                                                    [nan]   
3                                                    [nan]   
4                                          [nan, nan, nan]   
...                                                    ...   
1870743                                    [nan, nan, nan]   
1870744  [nan, nan, nan, nan, nan, nan, nan, nan, nan, ...   
1870745                                    [nan, nan, nan]   
1870746                                    [nan, nan, nan]   
1870747                                              [nan]   

                                          canonical_smiles mutation    year  \
0        CN[C@@H](C)C(=O)N[C@H](C(=O)N1CC[C@H](C)[C@H]1...      NaN  2020.0   
1        O=C1OC2(c3ccccc31)c1cc(I)c(O)c(I)c1Oc1c2cc(I)c...      NaN  2020.0   
2                  O=c1cc(-c2ccc(O)c(O)c2)oc2cc(O)cc(O)c12      NaN  2020.0   
3                       CCN(CCO)CCCC(C)Nc1ccnc2cc(Cl)ccc12      NaN  2020.0   
4

In [49]:
# Check how many bioactivity datapoints in ChEMBL are tested in mutants
chembl_dataset_new_round[~chembl_dataset_new_round['target_id'].str.contains('WT')]

target_id      chembl_id  \
257        A0A045ISB3_F161S  CHEMBL4637373   
258        A0A045ISB3_F161S  CHEMBL4637459   
259        A0A045ISB3_F161S  CHEMBL4638011   
260        A0A045ISB3_F161S  CHEMBL4647367   
261        A0A045ISB3_I203A  CHEMBL4637373   
...                     ...            ...   
1868112         Q9YQ12_V82F  CHEMBL3797292   
1868113         Q9YQ12_V82F  CHEMBL3799941   
1868114         Q9YQ12_V82F  CHEMBL4452050   
1870322  T1QYY3_G140S_Q148H  CHEMBL4779601   
1870323  T1QYY3_G140S_Q148H  CHEMBL4786078   

                                     assay_id  \
257                                 [1992590]   
258                                 [1992590]   
259                                 [1992590]   
260                                 [1992590]   
261                                 [1992587]   
...                                       ...   
1868112  [1576372, 1576372, 1822317, 1822317]   
1868113  [1576372, 1576372, 1822317, 1822317]   
1868114                             [1854379]   
1870322                             [2070418]   
1870323                             [2070418]   

                                      activity_id   accession  \
257                                    [20694452]  A0A045ISB3   
258                                    [20694450]  A0A045ISB3   
259                                    [20694453]  A0A045ISB3   
260                                    [20694451]  A0A045ISB3   
261                                    [20694440]  A0A045ISB3   
...                                           ...         ...   
1868112  [16572620, 16572621, 18916325, 18916333]      Q9YQ12   
1868113  [16572618, 16572619, 18916332, 18916379]      Q9YQ12   
1868114                                [19051909]      Q9YQ12   
1870322                                [22876490]      T1QYY3   
1870323                                [22876491]      T1QYY3   

                    pchembl_value      activity_comment  \
257                        [5.13]                 [nan]   
258                        [5.07]                 [nan]   
259                        [5.04]                 [nan]   
260                        [5.27]                 [nan]   
261                        [7.68]                 [nan]   
...                           ...                   ...   
1868112  [7.62, 7.89, 7.62, 7.89]  [nan, nan, nan, nan]   
1868113    [7.72, 8.0, 8.0, 7.72]  [nan, nan, nan, nan]   
1868114                    [8.77]                 [nan]   
1870322                    [7.77]                 [nan]   
1870323                     [8.1]                 [nan]   

                                          canonical_smiles     mutation  \
257          O=C(O)C(=O)Nc1ccc(C#Cc2ccc(C(F)(F)F)cc2Br)cc1        F161S   
258            O=C(O)C(=O)Nc1ccc(C#Cc2ccc(C(F)(F)F)cc2)cc1        F161S   
259           O=C(O)C(=O)Nc1ccc(C#Cc2ccc(C(F)(F)F)cc2F)cc1        F161S   
260          O=C(O)C(=O)Nc1ccc(C#Cc2ccc(C(F)(F)F)cc2Cl)cc1        F161S   
261          O=C(O)C(=O)Nc1ccc(C#Cc2ccc(C(F)(F)F)cc2Br)cc1        I203A   
...                                                    ...          ...   
1868112  COc1ccc(S(=O)(=O)N(CC(C)C)[C@@H](O)[C@H](Cc2cc...         V82F   
1868113  COc1ccc(S(=O)(=O)N(CC(C)C)[C@@H](O)[C@H](Cc2cc...         V82F   
1868114  COc1ccc(S(=O)(=O)N(CC(C)C)C[C@H](O)[C@@H](Cc2c...         V82F   
1870322  CN(C)C(=O)C(=O)NC12CCC(CC1)Cn1c2nc(C(=O)NCc2cc...  G140S,Q148H   
1870323  Cc1cc(CNC(=O)c2nc3n(c(=O)c2O)CC2CCC3(NC(=O)C(=...  G140S,Q148H   

           year                                           sequence  \
257      2020.0  MAVRELPGAWNFRDVADTATALRPGRLFRSSELSRLDDAGRATLRR...   
258      2020.0  MAVRELPGAWNFRDVADTATALRPGRLFRSSELSRLDDAGRATLRR...   
259      2020.0  MAVRELPGAWNFRDVADTATALRPGRLFRSSELSRLDDAGRATLRR...   
260      2020.0  MAVRELPGAWNFRDVADTATALRPGRLFRSSELSRLDDAGRATLRR...   
261      2020.0  MAVRELPGAWNFRDVADTATALRPGRLFRSSELSRLDDAGRATLRR...   
...         ...                                                ..

In [50]:
# Check how many of those are tested on undefined mutants
chembl_dataset_new_round[chembl_dataset_new_round['target_id'].str.contains('MUTANT')]

target_id      chembl_id   assay_id activity_id accession  \
4394     A5Z252_MUTANT   CHEMBL222813  [1804226]  [18813644]    A5Z252   
4395     A5Z252_MUTANT  CHEMBL4276773  [1804226]  [18813639]    A5Z252   
4396     A5Z252_MUTANT  CHEMBL4278505  [1804226]  [18813633]    A5Z252   
4397     A5Z252_MUTANT  CHEMBL4281970  [1804226]  [18813635]    A5Z252   
4398     A5Z252_MUTANT  CHEMBL4282553  [1804226]  [18813637]    A5Z252   
...                ...            ...        ...         ...       ...   
1826325  Q9UM73_MUTANT  CHEMBL3669148  [1640332]  [17701115]    Q9UM73   
1826326  Q9UM73_MUTANT  CHEMBL3669149  [1640332]  [17701116]    Q9UM73   
1826327  Q9UM73_MUTANT  CHEMBL3669150  [1640332]  [17701117]    Q9UM73   
1826328  Q9UM73_MUTANT  CHEMBL3669151  [1640332]  [17701118]    Q9UM73   
1826329  Q9UM73_MUTANT  CHEMBL3928179  [1640332]  [17623333]    Q9UM73   

        pchembl_value activity_comment  \
4394           [8.12]            [nan]   
4395           [7.44]            [nan]   
4396           [6.04]            [nan]   
4397           [4.52]            [nan]   
4398           [6.44]            [nan]   
...               ...              ...   
1826325        [7.89]            [nan]   
1826326        [7.37]            [nan]   
1826327        [6.37]            [nan]   
1826328        [7.54]            [nan]   
1826329        [7.85]            [nan]   

                                          canonical_smiles mutation    year  \
4394     CC(=O)N[C@@H]1[C@@H](NC(=N)N)C=C(C(=O)O)O[C@H]...    H274Y  2018.0   
4395     CCC(CC)O[C@@H]1C=C(C(=O)O)C[C@H](NCc2ccc(Sc3cc...    H274Y  2018.0   
4396     CCC(CC)O[C@@H]1C=C(C(=O)O)C[C@H](NCc2ccc(Cc3cc...    H274Y  2018.0   
4397     CCC(CC)O[C@@H]1C=C(C(=O)O)C[C@H](NCc2cccc(OC)c...    H274Y  2018.0   
4398     CCC(CC)O[C@@H]1C=C(C(=O)O)C[C@H](NCc2ccc(Oc3cc...    H274Y  2018.0   
...                                                    ...      ...     ...   
1826325  COc1ncc(F)cc1-c1c(C(N)=O)sc2cnc(Nc3cc(C4CCOCC4...   L1196M  2015.0   
1826326  COc1ncccc1-c1c(C(N)=O)sc2cnc(Nc3cccc(N4CCN(C)C...   L1196M  2015.0   
1826327  COc1ccccc1-c1c(C(N)=O)sc2cnc(Nc3cc4c(cc3OC(C)C...   L1196M  2015.0   
1826328  COc1ccc(F)cc1-c1c(CO)sc2cnc(Nc3ccc(N4CCN(C)CC4...   L1196M  2015.0   
1826329  COc1nc(C)ccc1-c1c(CO)sc2cnc(Nc3cc(C)c(N4CCN(C)...   L1196M  2015.0   

                                                  sequence  \
4394     MNPNQKIITIGSICMVIGIVSLMLQIGNMISIWVSHSIQTGNQRQA...   
4395     MNPNQKIITIGSICMVIGIVSLMLQIGNMISIWVSHSIQTGNQRQA...   
4396     MNPNQKIITIGSICMVIGIVSLMLQIGNMISIWVSHSIQTGNQRQA...   
4397     MNPNQKIITIGSICMVIGIVSLMLQIGNMISIWVSHSIQTGNQRQA...   
4398     MNPNQKIITIGSICMVIGIVSLMLQIGNMISIWVSHSIQTGNQRQA...   
...                                                    ...   
1826325  MGAIGLLWLLPLLLSTAAVGSGMGTGQRAGSPAAGPPLQPREPLSY...   
1826326  MGAIGLLWLLPLLLSTAAVGSGMGTGQRAGSPAAGPPLQPREPLSY...   
1826327  MGAIGLLWLLPLLLSTAAVGSGMGTGQRAGSPAAGPPLQPREPLSY...   
1826328  MGAIGLLWLLPLLLSTAAVGSGMGTGQRAGSPAAGPPLQPREPLSY...   
1826329  MGAIGLLWLLPLLLSTAAVGSGMGTGQRAGSPAAGPPLQPREPLSY...   

         pchembl_value_Mean Activity_class  
4394                   8.12            NaN  
4395                   7.44            NaN  
4396                   6.04            NaN  
4397                   4.52            NaN  
4398                   6.44            NaN  
...                     ...            ...  
1826325                7.89            NaN  
1826326                7.37            NaN  
1826327                6.37            NaN  
1826328                7.54            NaN  
1826329                7.85            NaN  

[736 rows x 13 columns]

In [51]:
# Concatenate newly annotated ChEMBL bioactivity data with non-ChEMBL part of Papyrus (only targets with at least
# one annotated variant)
chembl_papyrus_bioactivity_dataset_new_round = combine_chembl_papyrus_mutants(chembl_version, papyrus_version, papyrus_flavor, 1_000_000, annotation_round)
chembl_papyrus_bioactivity_dataset_new_round

C:\Users\gorostiolam\AppData\Local\Continuum\anaconda3\envs\mutants-in-pcm\lib\site-packages\ipykernel_launcher.py:3: DtypeWarning: Columns (9) have mixed types.Specify dtype option on import or set low_memory=False.
  This is separate from the ipykernel package so we can avoid doing imports until


CID    connectivity         target_id  \
0                      CHEMBL4637373  LUVQLVJTOZVLDB  A0A045ISB3_F161S   
1                      CHEMBL4637459  ALIJNJWQOZKBPP  A0A045ISB3_F161S   
2                      CHEMBL4638011  IPTNWUSXZLVWFT  A0A045ISB3_F161S   
3                      CHEMBL4647367  CXXIUJZSLPZZIM  A0A045ISB3_F161S   
4                      CHEMBL4637373  LUVQLVJTOZVLDB  A0A045ISB3_I203A   
...                              ...             ...               ...   
481316  Christmann2016.compound.1413  VMJFTOSOFDEKTM     P08581_N1100Y   
481317  Christmann2016.compound.1413  VMJFTOSOFDEKTM     P08581_Y1230C   
481318  Christmann2016.compound.1413  VMJFTOSOFDEKTM     P08581_Y1230H   
481319  Christmann2016.compound.1640  WCIGMFCFPXZRMQ     Q5S007_A2016T   
481320  Christmann2016.compound.1392  YEVOZZZLKJKCCD     Q5S007_A2016T   

                         AID   accession  pchembl_value_Mean  \
0                  [1992590]  A0A045ISB3               5.130   
1                  [1992590]  A0A045ISB3               5.070   
2                  [1992590]  A0A045ISB3               5.040   
3                  [1992590]  A0A045ISB3               5.270   
4                  [1992587]  A0A045ISB3               7.680   
...                      ...         ...                 ...   
481316  Christmann2016.assay      P08581               8.699   
481317  Christmann2016.assay      P08581               9.000   
481318  Christmann2016.assay      P08581               9.222   
481319  Christmann2016.assay      Q5S007               7.091   
481320  Christmann2016.assay      Q5S007               6.813   

                                                   SMILES  \
0           O=C(O)C(=O)Nc1ccc(C#Cc2ccc(C(F)(F)F)cc2Br)cc1   
1             O=C(O)C(=O)Nc1ccc(C#Cc2ccc(C(F)(F)F)cc2)cc1   
2            O=C(O)C(=O)Nc1ccc(C#Cc2ccc(C(F)(F)F)cc2F)cc1   
3           O=C(O)C(=O)Nc1ccc(C#Cc2ccc(C(F)(F)F)cc2Cl)cc1   
4           O=C(O)C(=O)Nc1ccc(C#Cc2ccc(C(F)(F)F)cc2Br)cc1   
...                                                   ...   
481316  Cn1cc(-c2cnc3ccc4c(cc(CS(=O)(=O)NCc5ncccc5)cc4...   
481317  Cn1cc(-c2cnc3ccc4c(cc(CS(=O)(=O)NCc5ncccc5)cc4...   
481318  Cn1cc(-c2cnc3ccc4c(cc(CS(=O)(=O)NCc5ncccc5)cc4...   
481319     O=C(Nc1cnccc1)c1cc(-c2ccnc(F)c2)ccc1OCc1ccccc1   
481320         CNc1nc(Nc2c(OC)cc(C(=O)N3CCOCC3)cc2)ncc1Cl   

                                                 sequence  \
0       MAVRELPGAWNFRDVADTATALRPGRLFRSSELSRLDDAGRATLRR...   
1       MAVRELPGAWNFRDVADTATALRPGRLFRSSELSRLDDAGRATLRR...   
2       MAVRELPGAWNFRDVADTATALRPGRLFRSSELSRLDDAGRATLRR...   
3       MAVRELPGAWNFRDVADTATALRPGRLFRSSELSRLDDAGRATLRR...   
4       MAVRELPGAWNFRDVADTATALRPGRLFRSSELSRLDDAGRATLRR...   
...                                                   ...   
481316  MKAPAVLAPGILVLLFTLVQRSNGECKEALAKSEMNVNMKYQLPNF...   
481317  MKAPAVLAPGILVLLFTLVQRSNGECKEALAKSEMNVNMKYQLPNF...   
481318  MKAPAVLAPGILVLLFTLVQRSNGECKEALAKSEMNVNMKYQLPNF...   
481319  MASGSCQGCEEDEETLKKLIVRLNNVQEGKQIETLVQILEDLLVFT...   
481320  MASGSCQGCEEDEETLKKLIVRLNNVQEGKQIETLVQILEDLLVFT...   

                            source Activity_class    Year  
0                         ChEMBL31            NaN  2020.0  
1                         ChEMBL31            NaN  2020.0  
2                         ChEMBL31            NaN  2020.0  
3                         ChEMBL31            NaN  2020.0  
4                         ChEMBL31            NaN  2020.0  
...                            ...            ...     ...  
481316  Papyrus05.5_Christmann2016            NaN  2011.0  
481317  Papyrus05.5_Christmann2016            NaN  2011.0  
481318  Papyrus05.5_Christmann2016            NaN  2011.0  
481319  Papyrus05.5_Christmann2016            NaN  2012.0  
481320  Papyrus05.5_Christmann2016            NaN  2012.0  

[481321 rows x 11 columns]

In [52]:
annotated_data_new_round = merge_chembl_papyrus_mutants(chembl_version, papyrus_version, papyrus_flavor, 1_000_000, annotation_round)
annotated_data_new_round

target_id    connectivity  pchembl_value_Mean  \
0       A0A045ISB3_F161S  ALIJNJWQOZKBPP                5.07   
1       A0A045ISB3_F161S  CXXIUJZSLPZZIM                5.27   
2       A0A045ISB3_F161S  IPTNWUSXZLVWFT                5.04   
3       A0A045ISB3_F161S  LUVQLVJTOZVLDB                5.13   
4       A0A045ISB3_I203A  ALIJNJWQOZKBPP                7.37   
...                  ...             ...                 ...   
455834         Q9YQ12_WT  ZYLARFCKPNSSDA                6.17   
455835         Q9YQ12_WT  ZYLFWZWRSRMGBQ               10.10   
455836         Q9YQ12_WT  ZZCUJLFFPCGLAL                5.51   
455837         Q9YQ12_WT  ZZGMOZUZSAJAML                6.26   
455838         Q9YQ12_WT  ZZOKRSQSLSCEAE                7.52   

         Activity_class_consensus    source  \
0       Series([], dtype: object)  ChEMBL31   
1       Series([], dtype: object)  ChEMBL31   
2       Series([], dtype: object)  ChEMBL31   
3       Series([], dtype: object)  ChEMBL31   
4       Series([], dtype: object)  ChEMBL31   
...                           ...       ...   
455834  Series([], dtype: object)  ChEMBL31   
455835  Series([], dtype: object)  ChEMBL31   
455836  Series([], dtype: object)  ChEMBL31   
455837  Series([], dtype: object)  ChEMBL31   
455838  Series([], dtype: object)  ChEMBL31   

                                                   SMILES            CID  \
0             O=C(O)C(=O)Nc1ccc(C#Cc2ccc(C(F)(F)F)cc2)cc1  CHEMBL4637459   
1           O=C(O)C(=O)Nc1ccc(C#Cc2ccc(C(F)(F)F)cc2Cl)cc1  CHEMBL4647367   
2            O=C(O)C(=O)Nc1ccc(C#Cc2ccc(C(F)(F)F)cc2F)cc1  CHEMBL4638011   
3           O=C(O)C(=O)Nc1ccc(C#Cc2ccc(C(F)(F)F)cc2Br)cc1  CHEMBL4637373   
4             O=C(O)C(=O)Nc1ccc(C#Cc2ccc(C(F)(F)F)cc2)cc1  CHEMBL4637459   
...                                                   ...            ...   
455834  CC(C)[C@H](NC(=O)COc1ccccc1)C(=O)N[C@@H](Cc1cc...  CHEMBL3331342   
455835  Nc1cccc(CN2C(=O)N(Cc3ccc4cn[nH]c4c3)C(Cc3ccccc...  CHEMBL2296983   
455836  CC1(C)CC[C@]2(C(=O)NCCCCCC(=O)NCC(=O)O)CC[C@]3...  CHEMBL4530451   
455837  CC(C)c1nc(CN(C(=O)N[C@H](C(=O)N[C@H](CC[C@H](C...  CHEMBL3115161   
455838  CC1(C)CC(=O)C2=C(C1)Nc1ccccc1N(C(=O)C(=O)N1c3c...  CHEMBL2180490   

         accession                                           sequence    Year  \
0       A0A045ISB3  MAVRELPGAWNFRDVADTATALRPGRLFRSSELSRLDDAGRATLRR...  2020.0   
1       A0A045ISB3  MAVRELPGAWNFRDVADTATALRPGRLFRSSELSRLDDAGRATLRR...  2020.0   
2       A0A045ISB3  MAVRELPGAWNFRDVADTATALRPGRLFRSSELSRLDDAGRATLRR...  2020.0   
3       A0A045ISB3  MAVRELPGAWNFRDVADTATALRPGRLFRSSELSRLDDAGRATLRR...  2020.0   
4       A0A045ISB3  MAVRELPGAWNFRDVADTATALRPGRLFRSSELSRLDDAGRATLRR...  2020.0   
...            ...                                                ...     ...   
455834      Q9YQ12  PQITLWQRPFVTIKIEGQLKEALLDTGADDTVLEEMNLPGRWKPKM...  2014.0   
455835      Q9YQ12  PQITLWQRPFVTIKIEGQLKEALLDTGADDTVLEEMNLPGRWKPKM...  2013.0   
455836      Q9YQ12  PQITLWQRPFVTIKIEGQLKEALLDTGADDTVLEEMNLPGRWKPKM...  2019.0   
455837      Q9YQ12  PQITLWQRPFVTIKIEGQLKEALLDTGADDTVLEEMNLPGRWKPKM...  2014.0   
455838      Q9YQ12  PQITLWQRPFVTIKIEGQLKEALLDTGADDTVLEEMNLPGRWKPKM...  2012.0   

               UniProtID                        Organism HGNC_symbol  
0       A0A045ISB3_MYCTX      Mycobacterium tuberculosis         NaN  
1       A0A045ISB3_MYCTX      Mycobacterium tuberculosis         NaN  
2       A0A045ISB3_MYCTX      Mycobacterium tuberculosis         NaN  
3       A0A045ISB3_MYCTX      Mycobacterium tuberculosis         NaN  
4       A0A045ISB3_MYCTX      Mycobacterium tuberculosis         NaN  
...                  ...                             ...         ...  
455834      Q9YQ12_9HIV1  Human immunodeficiency virus 1         NaN  
455835      Q9YQ12_9HIV1  Human immunodeficiency virus 1         NaN  
455836      Q9YQ12_9HIV1  Human immunodeficiency virus 1         NaN  
455837      Q9YQ12_9HIV1  Human immunodeficiency virus 1         NaN  
455838      Q9

In [53]:
# Check how many bioactivity datapoints in ChEMBL + Papyrus are tested in mutants
annotated_data_new_round[~annotated_data_new_round['target_id'].str.contains('WT')]

target_id    connectivity  pchembl_value_Mean  \
0       A0A045ISB3_F161S  ALIJNJWQOZKBPP               5.070   
1       A0A045ISB3_F161S  CXXIUJZSLPZZIM               5.270   
2       A0A045ISB3_F161S  IPTNWUSXZLVWFT               5.040   
3       A0A045ISB3_F161S  LUVQLVJTOZVLDB               5.130   
4       A0A045ISB3_I203A  ALIJNJWQOZKBPP               7.370   
...                  ...             ...                 ...   
454742       Q9YQ12_V82A  SBBYGVNSLWBXFB               7.455   
454743       Q9YQ12_V82A  XHKGWTXTLQTTQA               7.400   
454744       Q9YQ12_V82F  MEWAZRJLRMEJDV               8.770   
454745       Q9YQ12_V82F  SBBYGVNSLWBXFB               7.860   
454746       Q9YQ12_V82F  XHKGWTXTLQTTQA               7.755   

         Activity_class_consensus    source  \
0       Series([], dtype: object)  ChEMBL31   
1       Series([], dtype: object)  ChEMBL31   
2       Series([], dtype: object)  ChEMBL31   
3       Series([], dtype: object)  ChEMBL31   
4       Series([], dtype: object)  ChEMBL31   
...                           ...       ...   
454742  Series([], dtype: object)  ChEMBL31   
454743  Series([], dtype: object)  ChEMBL31   
454744  Series([], dtype: object)  ChEMBL31   
454745  Series([], dtype: object)  ChEMBL31   
454746  Series([], dtype: object)  ChEMBL31   

                                                   SMILES            CID  \
0             O=C(O)C(=O)Nc1ccc(C#Cc2ccc(C(F)(F)F)cc2)cc1  CHEMBL4637459   
1           O=C(O)C(=O)Nc1ccc(C#Cc2ccc(C(F)(F)F)cc2Cl)cc1  CHEMBL4647367   
2            O=C(O)C(=O)Nc1ccc(C#Cc2ccc(C(F)(F)F)cc2F)cc1  CHEMBL4638011   
3           O=C(O)C(=O)Nc1ccc(C#Cc2ccc(C(F)(F)F)cc2Br)cc1  CHEMBL4637373   
4             O=C(O)C(=O)Nc1ccc(C#Cc2ccc(C(F)(F)F)cc2)cc1  CHEMBL4637459   
...                                                   ...            ...   
454742  COc1ccc(S(=O)(=O)N(CC(C)C)[C@@H](O)[C@H](Cc2cc...  CHEMBL3799941   
454743  COc1ccc(S(=O)(=O)N(CC(C)C)[C@@H](O)[C@H](Cc2cc...  CHEMBL3797292   
454744  COc1ccc(S(=O)(=O)N(CC(C)C)C[C@H](O)[C@@H](Cc2c...  CHEMBL4452050   
454745  COc1ccc(S(=O)(=O)N(CC(C)C)[C@@H](O)[C@H](Cc2cc...  CHEMBL3799941   
454746  COc1ccc(S(=O)(=O)N(CC(C)C)[C@@H](O)[C@H](Cc2cc...  CHEMBL3797292   

         accession                                           sequence    Year  \
0       A0A045ISB3  MAVRELPGAWNFRDVADTATALRPGRLFRSSELSRLDDAGRATLRR...  2020.0   
1       A0A045ISB3  MAVRELPGAWNFRDVADTATALRPGRLFRSSELSRLDDAGRATLRR...  2020.0   
2       A0A045ISB3  MAVRELPGAWNFRDVADTATALRPGRLFRSSELSRLDDAGRATLRR...  2020.0   
3       A0A045ISB3  MAVRELPGAWNFRDVADTATALRPGRLFRSSELSRLDDAGRATLRR...  2020.0   
4       A0A045ISB3  MAVRELPGAWNFRDVADTATALRPGRLFRSSELSRLDDAGRATLRR...  2020.0   
...            ...                                                ...     ...   
454742      Q9YQ12  PQITLWQRPFVTIKIEGQLKEALLDTGADDTVLEEMNLPGRWKPKM...  2016.0   
454743      Q9YQ12  PQITLWQRPFVTIKIEGQLKEALLDTGADDTVLEEMNLPGRWKPKM...  2016.0   
454744      Q9YQ12  PQITLWQRPFVTIKIEGQLKEALLDTGADDTVLEEMNLPGRWKPKM...  2019.0   
454745      Q9YQ12  PQITLWQRPFVTIKIEGQLKEALLDTGADDTVLEEMNLPGRWKPKM...  2016.0   
454746      Q9YQ12  PQITLWQRPFVTIKIEGQLKEALLDTGADDTVLEEMNLPGRWKPKM...  2016.0   

               UniProtID                        Organism HGNC_symbol  
0       A0A045ISB3_MYCTX      Mycobacterium tuberculosis         NaN  
1       A0A045ISB3_MYCTX      Mycobacterium tuberculosis         NaN  
2       A0A045ISB3_MYCTX      Mycobacterium tuberculosis         NaN  
3       A0A045ISB3_MYCTX      Mycobacterium tuberculosis         NaN  
4       A0A045ISB3_MYCTX      Mycobacterium tuberculosis         NaN  
...                  ...                             ...         ...  
454742      Q9YQ12_9HIV1  Human immunodeficiency virus 1         NaN  
454743      Q9YQ12_9HIV1  Human immunodeficiency virus 1         NaN  
454744      Q9YQ12_9HIV1  Human immunodeficiency virus 1         NaN  
454745      Q9YQ12_9HIV1  Human immunodeficiency virus 1         NaN  
454746      Q9

In [54]:
# Check how many bioactivity datapoints are tested in undefined mutants
annotated_data_new_round[annotated_data_new_round['target_id'].str.contains('MUTANT')]

target_id    connectivity  pchembl_value_Mean  \
1886    A5Z252_MUTANT  ARAIBEBZBOPLMB                8.12   
1887    A5Z252_MUTANT  BWDGROPAFQXXFI                6.44   
1888    A5Z252_MUTANT  DSRHDFMYUHIMKS                4.57   
1889    A5Z252_MUTANT  ITKUZBOVXCDIPW                6.14   
1890    A5Z252_MUTANT  MCRWKUDKDJFWEE                7.44   
...               ...             ...                 ...   
451580  Q9UM73_MUTANT  OFVDSACHDVLZDS                8.52   
451581  Q9UM73_MUTANT  QFALLVRHZMVEMI                7.24   
451582  Q9UM73_MUTANT  SWRGFJBGNAEJOY                7.54   
451583  Q9UM73_MUTANT  URHVQJZEFHMQBZ                7.27   
451584  Q9UM73_MUTANT  YUAGJCVZVLZRDP                8.05   

         Activity_class_consensus    source  \
1886    Series([], dtype: object)  ChEMBL31   
1887    Series([], dtype: object)  ChEMBL31   
1888    Series([], dtype: object)  ChEMBL31   
1889    Series([], dtype: object)  ChEMBL31   
1890    Series([], dtype: object)  ChEMBL31   
...                           ...       ...   
451580  Series([], dtype: object)  ChEMBL31   
451581  Series([], dtype: object)  ChEMBL31   
451582  Series([], dtype: object)  ChEMBL31   
451583  Series([], dtype: object)  ChEMBL31   
451584  Series([], dtype: object)  ChEMBL31   

                                                   SMILES            CID  \
1886    CC(=O)N[C@@H]1[C@@H](NC(=N)N)C=C(C(=O)O)O[C@H]...   CHEMBL222813   
1887    CCC(CC)O[C@@H]1C=C(C(=O)O)C[C@H](NCc2ccc(Oc3cc...  CHEMBL4282553   
1888    CCC(CC)O[C@@H]1C=C(C(=O)O)C[C@H](NCc2ccc(-c3cc...  CHEMBL4285029   
1889    CCc1ccc(CN[C@H]2CC(C(=O)O)=C[C@@H](OC(CC)CC)[C...  CHEMBL4285767   
1890    CCC(CC)O[C@@H]1C=C(C(=O)O)C[C@H](NCc2ccc(Sc3cc...  CHEMBL4276773   
...                                                   ...            ...   
451580  COc1ncccc1-c1c(C(N)=O)sc2cnc(Nc3cc(C)c(N4CCN(C...  CHEMBL3669139   
451581  COc1ccccc1-c1c(C(N)=O)sc2cnc(Nc3ccc(NC4CCOCC4)...  CHEMBL3669147   
451582  COc1ccc(F)cc1-c1c(CO)sc2cnc(Nc3ccc(N4CCN(C)CC4...  CHEMBL3669151   
451583  COc1ccccc1-c1c(C(N)=O)sc2cnc(Nc3ccc(-c4cnn(C)c...  CHEMBL3669143   
451584  COc1ccccc1-c1c(C(N)=O)sc2cnc(Nc3ccc(CN4CCN(C)C...  CHEMBL3669146   

       accession                                           sequence    Year  \
1886      A5Z252  MNPNQKIITIGSICMVIGIVSLMLQIGNMISIWVSHSIQTGNQRQA...  2018.0   
1887      A5Z252  MNPNQKIITIGSICMVIGIVSLMLQIGNMISIWVSHSIQTGNQRQA...  2018.0   
1888      A5Z252  MNPNQKIITIGSICMVIGIVSLMLQIGNMISIWVSHSIQTGNQRQA...  2018.0   
1889      A5Z252  MNPNQKIITIGSICMVIGIVSLMLQIGNMISIWVSHSIQTGNQRQA...  2018.0   
1890      A5Z252  MNPNQKIITIGSICMVIGIVSLMLQIGNMISIWVSHSIQTGNQRQA...  2018.0   
...          ...                                                ...     ...   
451580    Q9UM73  MGAIGLLWLLPLLLSTAAVGSGMGTGQRAGSPAAGPPLQPREPLSY...  2015.0   
451581    Q9UM73  MGAIGLLWLLPLLLSTAAVGSGMGTGQRAGSPAAGPPLQPREPLSY...  2015.0   
451582    Q9UM73  MGAIGLLWLLPLLLSTAAVGSGMGTGQRAGSPAAGPPLQPREPLSY...  2015.0   
451583    Q9UM73  MGAIGLLWLLPLLLSTAAVGSGMGTGQRAGSPAAGPPLQPREPLSY...  2015.0   
451584    Q9UM73  MGAIGLLWLLPLLLSTAAVGSGMGTGQRAGSPAAGPPLQPREPLSY...  2015.0   

           UniProtID                                        Organism  \
1886    A5Z252_9INFA  Influenza A virus (A/Turkey/651242/2006(H5N1))   
1887    A5Z252_9INFA  Influenza A virus (A/Turkey/651242/2006(H5N1))   
1888    A5Z252_9INFA  Influenza A virus (A/Turkey/651242/2006(H5N1))   
1889    A5Z252_9INFA  Influenza A virus (A/Turkey/651242/2006(H5N1))   
1890    A5Z252_9INFA  Influenza A virus (A/Turkey/651242/2006(H5N1))   
...              ...                                             ...   
451580     ALK_HUMAN                            Homo sapiens (Human)   
451581     ALK_HUMAN                            Homo sapiens (Human)   
451582     ALK_HUMAN                            Homo sapiens (Human)   
451583     ALK_HUMAN                            Homo sapiens (Human)   
451584     ALK_HUMAN                            Homo sapiens

In [55]:
# Check the distribution of per source of mutant bioactivity datapoints
annotated_data_new_round[~annotated_data_new_round['target_id'].str.contains('WT')].groupby(['source'])['pchembl_value_Mean'].count()

source
ChEMBL31                               22992
Papyrus05.5_Christmann2016               672
Papyrus05.5_Christmann2016;ChEMBL31     1422
Name: pchembl_value_Mean, dtype: int64

#### Quickly check EGFR as case study

##### Check how many deletions are there

In [56]:
# Extract EGFR deletions from false positives and negatives from round 1 
false_negative_deletions = false_negatives[false_negatives['rejection_flag'] == 'original_deletion']
false_positive_deletions = false_positives[false_positives['reason'] == 'missing deletion']
egfr_deletions = pd.concat([false_positive_deletions[false_positive_deletions['accession']=='P00533'][['assay_id',
                                                                                                       'accession']],
                            false_negative_deletions[false_negative_deletions['accession']=='P00533'][['assay_id',
                                                                                                  'accession']]])
egfr_deletions['deletion'] = True

# Attach deletion flags to round 2 assays (with updated target_id)
chembl_assays_with_egfr_deletions = pd.merge(new_round_assays[new_round_assays['accession']=='P00533'], egfr_deletions,
                                             how='left',on=['assay_id','accession']).fillna(False)

# Check how many assays (from ChEMBL) have hidden deletions per EGFR annotated mutation
chembl_assays_with_egfr_deletions.groupby(['target_id','deletion'])['assay_id'].count()

target_id                 deletion
P00533_A750P              False          2
                          True          26
P00533_C797S              False          7
P00533_C797S_L858R        False          2
P00533_G719C              False         30
P00533_G719S              False         30
P00533_L858R              False        176
                          True           1
P00533_L861Q              False         58
P00533_P753S              False          2
                          True          25
P00533_T790M              False        106
                          True          15
P00533_T790M_C797S        False          2
                          True           5
P00533_T790M_C797S_L858R  False         30
P00533_T790M_L858M        False          1
P00533_T790M_L858R        False        293
P00533_T790M_L861R        False          1
P00533_WT                 False       1810
Name: assay_id, dtype: int64

In [57]:
# Attach ChEMBL data to round 2 assays with deletion flags
chembl_data = obtain_chembl_data(chembl_version)
# Not possible to ChEMBL + Papyrus round 2 (annotated_data_new_round) because it does not have assay_id variable anymore
chembl_data_with_egfr_deletions = pd.merge(chembl_data[chembl_data['accession']=='P00533'][['assay_id','accession',
                                                                                            'chembl_id','mutation']],
                                           chembl_assays_with_egfr_deletions[['assay_id','accession','target_id',
                                                                              'deletion']],
                                           how='left',on=['assay_id','accession']).fillna(False)
# Check how many activity datapoints (from ChEMBL) have hidden deletions per EGFR annotated mutation
chembl_data_with_egfr_deletions.groupby(['target_id','deletion'])['chembl_id'].count()

C:\Users\gorostiolam\AppData\Local\Continuum\anaconda3\envs\mutants-in-pcm\lib\site-packages\ipykernel_launcher.py:2: DtypeWarning: Columns (4,6,7,11) have mixed types.Specify dtype option on import or set low_memory=False.
  


target_id                 deletion
P00533_A750P              False          20
                          True           62
P00533_C797S              False          30
P00533_C797S_L858R        False          34
P00533_G719C              False          91
P00533_G719S              False          82
P00533_L858R              False        1704
                          True            1
P00533_L861Q              False         901
P00533_P753S              False          18
                          True           58
P00533_T790M              False        1144
                          True          108
P00533_T790M_C797S        False           5
                          True           20
P00533_T790M_C797S_L858R  False         145
P00533_T790M_L858M        False          30
P00533_T790M_L858R        False        2757
P00533_T790M_L861R        False           1
P00533_WT                 False       14100
Name: chembl_id, dtype: int64